include/maths/CBoostedTreeHyperparameters.h

```diff
@@ -0,0 +1,242 @@
+            RESTORE(REGULARIZATION_TREE_SIZE_PENALTY_MULTIPLIER_TAG,
+                    core::CPersistUtils::restore(REGULARIZATION_TREE_SIZE_PENALTY_MULTIPLIER_TAG,
+                                                 m_TreeSizePenaltyMultiplier, traverser))
+            RESTORE(REGULARIZATION_LEAF_WEIGHT_PENALTY_MULTIPLIER_TAG,
+                    core::CPersistUtils::restore(REGULARIZATION_LEAF_WEIGHT_PENALTY_MULTIPLIER_TAG,
+                                                 m_LeafWeightPenaltyMultiplier, traverser))
+            RESTORE(REGULARIZATION_SOFT_TREE_DEPTH_LIMIT_TAG,
+                    core::CPersistUtils::restore(REGULARIZATION_SOFT_TREE_DEPTH_LIMIT_TAG,
+                                                 m_SoftTreeDepthLimit, traverser))
+            RESTORE(REGULARIZATION_SOFT_TREE_DEPTH_TOLERANCE_TAG,
+                    core::CPersistUtils::restore(REGULARIZATION_SOFT_TREE_DEPTH_TOLERANCE_TAG,
+                                                 m_SoftTreeDepthTolerance, traverser))
+        } while (traverser.next());
+        return true;
+    };
+
+public:
+    static const std::string REGULARIZATION_DEPTH_PENALTY_MULTIPLIER_TAG;
+    static const std::string REGULARIZATION_TREE_SIZE_PENALTY_MULTIPLIER_TAG;
+    static const std::string REGULARIZATION_LEAF_WEIGHT_PENALTY_MULTIPLIER_TAG;
+    static const std::string REGULARIZATION_SOFT_TREE_DEPTH_LIMIT_TAG;
+    static const std::string REGULARIZATION_SOFT_TREE_DEPTH_TOLERANCE_TAG;
+
+private:
+    using TOptionalDouble = boost::optional<double>;
+
+private:
+    static std::string toString(double x) { return std::to_string(x); }
+    static std::string toString(TOptionalDouble x) {
+        return x != boost::none ? toString(*x) : "null";
+    }
+
+private:
+    T m_DepthPenaltyMultiplier = T{};
+    T m_TreeSizePenaltyMultiplier = T{};
+    T m_LeafWeightPenaltyMultiplier = T{};
+    T m_SoftTreeDepthLimit = T{};
+    T m_SoftTreeDepthTolerance = T{};
+};
+
+template<typename T>
+const std::string CBoostedTreeRegularization<T>::REGULARIZATION_DEPTH_PENALTY_MULTIPLIER_TAG{
+    "regularization_depth_penalty_multiplier"};
+template<typename T>
+const std::string CBoostedTreeRegularization<T>::REGULARIZATION_TREE_SIZE_PENALTY_MULTIPLIER_TAG{
+    "regularization_tree_size_penalty_multiplier"};
+template<typename T>
+const std::string CBoostedTreeRegularization<T>::REGULARIZATION_LEAF_WEIGHT_PENALTY_MULTIPLIER_TAG{
+    "regularization_leaf_weight_penalty_multiplier"};
+template<typename T>
+const std::string CBoostedTreeRegularization<T>::REGULARIZATION_SOFT_TREE_DEPTH_LIMIT_TAG{
+    "regularization_soft_tree_depth_limit"};
+template<typename T>
+const std::string CBoostedTreeRegularization<T>::REGULARIZATION_SOFT_TREE_DEPTH_TOLERANCE_TAG{
+    "regularization_soft_tree_depth_tolerance"};
+
+//! \brief The algorithm parameters we'll directly optimise to improve test error.
+class CBoostedTreeHyperparameters {
```

Comment:
I don't think Windows will like this:
```suggestion
class MATHS_EXPORT CBoostedTreeHyperparameters {
```